Gems/Atom/RHI/Metal/Code/Source/RHI/CommandQueue.cpp

```diff
@@ -89,61 +89,62 @@ namespace AZ
             auto& device = static_cast<Device&>(GetDevice());
             const ExecuteWorkRequest& request = static_cast<const ExecuteWorkRequest&>(rhiRequest);
             
-            bool haveEncodedList = false;
-            for (uint32_t commandListIdx = 0; commandListIdx < static_cast<uint32_t>(request.m_commandLists.size()); ++commandListIdx)
+            bool isCommitNeeded = request.m_signalFenceValue > 0 ||
+                                   request.m_scopeFencesToSignal.size() > 0 ||
+                                   request.m_swapChainsToPresent.size() > 0;
+            
+            for (uint32_t index = 0; index < aznumeric_cast<uint32_t>(request.m_commandLists.size()) && !isCommitNeeded; ++index)
             {
-                CommandList& commandList = *request.m_commandLists[commandListIdx];
+                CommandList& commandList = *request.m_commandLists[index];
                 if(commandList.IsEncoded())
                 {
-                    haveEncodedList = true;
-                    break;
+                    isCommitNeeded = true;
```

Comment:
Nit: Could `break` at this point to avoid checking `!isCommitNeeded` every loop iteration